internal-webapps/org.wso2.bfsi.authentication.endpoint/src/main/java/org/wso2/bfsi/authentication/endpoint/BFSIConsentServlet.java

```diff
@@ -0,0 +1,260 @@
+ * The servlet responsible for displaying the consent details in the auth UI flow.
+ */
+public class BFSIConsentServlet extends HttpServlet {
+
+    static BFSIAuthServletInterface bfsiAuthServletTK;
+    private static final long serialVersionUID = 6106269076132678046L;
+    private static Logger log = LoggerFactory.getLogger(BFSIConsentServlet.class);
+    private static final String BUNDLE = "org.wso2.bfsi.authentication.endpoint.i18n";
+    private static final ConsentManagementConfigParser parser = ConsentManagementConfigParser.getInstance();
+
+    @SuppressFBWarnings({"REQUESTDISPATCHER_FILE_DISCLOSURE", "TRUST_BOUNDARY_VIOLATION"})
+    // Suppressed content - obAuthServlet.getJSPPath()
+    // Suppression reason - False Positive : JSP path is hard coded and does not accept any user inputs, therefore it
+    //                      can be trusted
+    // Suppressed content - Encode.forJava(sessionDataKey)
+    // Suppression reason - False positive : sessionDataKey is encoded for Java which escapes untrusted characters
+    // Suppressed warning count - 2
+    @Override
+    public void doGet(HttpServletRequest originalRequest, HttpServletResponse response)
+            throws IOException, ServletException {
+        HttpServletRequest request = originalRequest;
+        setAuthExtension();
+
+        // get consent data
+        String sessionDataKey = request.getParameter(Constants.SESSION_DATA_KEY_CONSENT);
+        HttpResponse consentDataResponse = getConsentDataWithKey(sessionDataKey, getServletContext());
+        JSONObject dataSet = new JSONObject();
+        log.debug("HTTP response for consent retrieval" + consentDataResponse.toString());
+        try {
+            if (consentDataResponse.getStatusLine().getStatusCode() == HttpURLConnection.HTTP_MOVED_TEMP &&
+                    consentDataResponse.getLastHeader(Constants.LOCATION) != null) {
+                response.sendRedirect(consentDataResponse.getLastHeader(Constants.LOCATION).getValue());
+                return;
+            } else {
+                String retrievalResponse = IOUtils.toString(consentDataResponse.getEntity().getContent(),
+                        String.valueOf(StandardCharsets.UTF_8));
+                JSONObject data = new JSONObject(retrievalResponse);
+                String errorResponse = AuthenticationUtils.getErrorResponseForRedirectURL(data);
+                if (data.has(Constants.REDIRECT_URI) && StringUtils.isNotEmpty(errorResponse)) {
+                    URI errorURI = new URI(data.get(Constants.REDIRECT_URI).toString().concat(errorResponse));
+                    response.sendRedirect(errorURI.toString());
+                    return;
+                } else {
+                    dataSet = createConsentDataset(data, consentDataResponse.getStatusLine().getStatusCode());
+                }
+            }
+        } catch (IOException e) {
+            dataSet.put(Constants.IS_ERROR, "Exception occurred while retrieving consent data");
```

Comment:
Don't we need to log the exceptions?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/8/commits/5cab9c9a955547a16e4ab4f4f06ff55db6fb45c5